### PR TITLE
Disable claim review on errors

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Disable claim review button on errors

Screen Shot 2022-01-20 at 16 40 47

  # To Test

1. Load a claim on behalf of another account for which you have no funds to cover 100%
2. Go to approval step
* You should not be able to move forward

  # Background

Error due to lack of approval is still not captured, it'll be handled in another PR

### CODE REVIEW

**Comment by biocom** [comment]
Nice. One suggestion to make the validation error text more explicit:

> Insufficient {symbol} balance to cover the full investment amount. 
**Note:** When investing for another claim account, you can only invest the full (100%) amount and no partial amounts.

**Review by fairlighteth — APPROVED** [review]
Approve with a text suggestion.

**Comment by elena-zh** [comment]
Hey @alfetopito , to clarify: Will we disable 'Review' button when:
1.  claim for owner's account, and claim amount = 0 
2. token is not approved

Validation for claiming on behalf + when claim amount exceeds an investment amount works fine to me.

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx [hunk 1]
```diff
@@ -58,16 +58,16 @@ export default function InvestOption({ approveData, claim, optionIndex }: Invest
...
     setPercentage(_formatPercentage(calculatePercentage(balance, maxCost)))
-  }, [balance, maxCost, noBalance, optionIndex, updateInvestAmount])
+  }, [balance, maxCost, noBalance, optionIndex, updateInvestAmount, updateInvestError])
 
   // on input field change handler
   const onInputChange = useCallback(
     (value: string) => {
       setTypedValue(value)
-      setInputError('')
+      updateInvestError({ index: optionIndex, error: undefined })
```
Comment (W3stside) [inline src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:70]: nit: make a handleResetError or similar to clean these up and make them more declarative.

**Review by W3stside — APPROVED** [review]

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx (hunk 1 above)
Comment (alfetopito) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx]: Updated

**Comment by elena-zh** [comment]
> Hey @alfetopito , to clarify: Will we disable 'Review' button when:
> 
> 1. claim for owner's account, and claim amount = 0
> 2. token is not approved
> 
> Validation for claiming on behalf + when claim amount exceeds an investment amount works fine to me.

Covered in #2250

**Review by elena-zh — APPROVED** [review]

**Pull request merged**